examples/01_standalone_sdk/26_custom_visualizer.py

```diff
@@ -1,77 +1,267 @@
 """Custom Visualizer Example
 
 This example demonstrates how to create and use a custom visualizer by subclassing
-ConversationVisualizer. This approach provides:
+ConversationVisualizerBase. This approach provides:
 - Clean, testable code with class-based state management
 - Direct configuration (just pass the visualizer instance to visualizer parameter)
 - Reusable visualizer that can be shared across conversations
 
-This demonstrates how you can pass a ConversationVisualizer instance directly
+The SimpleVisualizer produces concise output showing:
+- LLM call completions with latency information
+- Tool execution steps with command/path details
+- Error messages
+
+This demonstrates how you can pass a ConversationVisualizerBase instance directly
 to the visualizer parameter for clean, reusable visualization logic.
 """
 
 import logging
 import os
+from collections.abc import Callable
 
 from pydantic import SecretStr
 
 from openhands.sdk import LLM, Conversation
 from openhands.sdk.conversation.visualizer import ConversationVisualizerBase
 from openhands.sdk.event import (
+    ActionEvent,
+    AgentErrorEvent,
     Event,
+    MessageEvent,
 )
+from openhands.sdk.llm.utils.metrics import Metrics
+from openhands.sdk.tool import Action
 from openhands.tools.preset.default import get_default_agent
 
 
-class MinimalVisualizer(ConversationVisualizerBase):
-    """A minimal visualizer that print the raw events as they occur."""
+def handles(event_type: type[Event]):
+    """Decorator to register a method as an event handler for a specific event type."""
+
+    def decorator(func):
+        func._handles_event_type = event_type
+        return func
+
+    return decorator
```

Comment:
I actually like this pattern, do you want to directly integrate this syntax into the default visualizer and/or even base class?